src/chains/ethereum/ethereum/tests/api/evm/evm.test.ts

```diff
@@ -79,90 +80,153 @@ describe("api", () => {
-      });
-
-      it("should mine a block on demand at the specified timestamp", async () => {
-        const startDate = new Date(2019, 3, 15);
-        const miningTimestamp = Math.floor(
-          new Date(2020, 3, 15).getTime() / 1000
-        );
-        const provider = await getProvider({ chain: { time: startDate } });
-        await provider.send("evm_mine", [miningTimestamp]);
-        const currentBlock = await provider.send("eth_getBlockByNumber", [
-          "latest"
-        ]);
-        assert.strictEqual(parseInt(currentBlock.timestamp), miningTimestamp);
-      });
-
-      it("should mine a block even when mining is stopped", async () => {
-        const provider = await getProvider();
-        const initialBlock = parseInt(await provider.send("eth_blockNumber"));
-        await provider.send("miner_stop");
-        await provider.send("evm_mine");
-        const currentBlock = parseInt(await provider.send("eth_blockNumber"));
-        assert.strictEqual(currentBlock, initialBlock + 1);
-      });
-
-      it("should mine a block when in interval mode", async () => {
-        const provider = await getProvider({ miner: { blockTime: 1000 } });
-        const initialBlock = parseInt(await provider.send("eth_blockNumber"));
-        await provider.send("evm_mine");
-        const currentBlock = parseInt(await provider.send("eth_blockNumber"));
-        assert.strictEqual(currentBlock, initialBlock + 1);
-      });
-
-      it("should mine a block when in interval mode even when mining is stopped", async () => {
-        const provider = await getProvider({ miner: { blockTime: 1000 } });
-        const initialBlock = parseInt(await provider.send("eth_blockNumber"));
-        await provider.send("miner_stop");
-        await provider.send("evm_mine");
-        const currentBlock = parseInt(await provider.send("eth_blockNumber"));
-        assert.strictEqual(currentBlock, initialBlock + 1);
-      });
-
-      it("should save the block before returning", async () => {
-        // slow down memdown's _batch function to consistently reproduce a past
-        // race condition where a block was mined and returned by evm_mine
-        // before it actually saved to the database. for history, the race
-        // condition issue is documented here:
-        // https://github.com/trufflesuite/ganache/issues/3060
-        const db = memdown();
-        let customBatchCalled = false;
-        db._batch = (...args) => {
-          setTimeout(() => {
-            customBatchCalled = true;
-            Reflect.apply(memdown.prototype._batch, db, args);
-          }, 20);
-        };
-
-        const options = { database: { db } };
-        const provider = await getProvider(options);
-        await provider.request({ method: "evm_mine", params: [{}] });
-
-        const block = await provider.request({
-          method: "eth_getBlockByNumber",
-          params: [`0x1`]
+      const providerOptions: EthereumProviderOptions[] = [
```

Comment:
I've added only one new test to this file. Here, I'm running all of the pre-existing tests with both the "eager" and "strict" mode